### PR TITLE
RPRBLND-2177: HybridPro add multimaterial support

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
Add multimaterial support to HybridPro.

### EFFECT OF CHANGE
Added support of multimaterial for Interactive render mode.

### TECHNICAL STEPS
Improved set_material_faces method in pyhybridpro.py module.
Enabled CONTEXT_CREATEPROP_HYBRID_ENABLE_PER_FACE_MATERIALS parameter. It should be set within context creation.
Added has_multimaterial_object variable for optimization purposes.

### CODE REVIEW

**Comment by VascoPi** [comment]
Discussed with HybridPro team, so let's wait for [PR-910](https://github.com/Radeon-Pro/RPRHybrid/pull/910).
It should bring some improvements related to the issues being noticed within testing.

**Comment by VascoPi** [comment]
Expected to be fixed with core 2.2.16

**Review by ARozghon — COMMENTED** [review]
waiting for core 2.2.16

**Review by DagerD — APPROVED** [review]
Tested. Works ok.

**Review by bsavery — APPROVED** [review]

**Comment by VascoPi** [comment]
1 sample
blender_1dtZcEMTSh

5 samples
blender_iJOob9gid5

20 samples
blender_jCqUynD7CM

**Review by bsavery — CHANGES_REQUESTED** [review]

File: src/rprblender/properties/render.py [hunk 1]
```diff
@@ -566,8 +567,15 @@ def init_rpr_context(self, rpr_context, is_final_engine=True, use_gl_interop=Fal
             context_props.extend([
                 pyrpr.CONTEXT_CREATEPROP_HYBRID_VERTEX_MEMORY_SIZE, vertex_mem_size,
                 pyrpr.CONTEXT_CREATEPROP_HYBRID_ACC_MEMORY_SIZE, acc_mem_size])
-               
-        context_props.append(0) # should be followed by 0
+
+        #  this functionality requires additional memory on
+        #  both CPU and GPU even when no per-face materials set in scene.
+        #  checking has_multimaterial_object before enable CONTEXT_CREATEPROP_HYBRID_ENABLE_PER_FACE_MATERIALS.
+        has_multimaterial_object = next((True for i in bpy.context.scene.objects if len(i.material_slots) > 1), False)
```
[the remark this answers is not in the document]
Comment (VascoPi) [reply src/rprblender/properties/render.py]: Improved condition for material slots for RPRContextHybridPro.

**Review by bsavery — APPROVED** [review]

**Review by ARozghon — APPROVED** [review]
Approved by QA

**Review by bnagirniak — APPROVED** [review]
Looks good

**Pull request merged**